build/i2c/src/lib.rs

```diff
@@ -200,9 +200,132 @@ struct I2cSensors {
     names: Option<Vec<String>>,
 }
 
-impl I2cSensors {
-    fn num_sensors(&self) -> usize {
-        self.temperature + self.power + self.current + self.voltage + self.speed
+#[derive(Debug)]
+struct I2cSensorsDescription {
+    // In all multimaps below, the value is the sensor ID. The same sensor ID
+    // can show up in multiple (including all!) of these maps.
+    //
+    // All sensors are guaranteed to be present in `bydevice` and `bykind`, but
+    // may not be present in the other maps (devices may or may not have a
+    // name/bus in app.toml).
+
+    // key: (device, kind)
+    bydevice: MultiMap<(String, Sensor), usize>,
+    // key: (device, name, kind)
+    byname: MultiMap<(String, String, Sensor), usize>,
+    // key: (device, bus, kind)
+    bybus: MultiMap<(String, String, Sensor), usize>,
+    // key: (device, bus, name, kind)
+    bybusname: MultiMap<(String, String, String, Sensor), usize>,
+    // key: kind
+    bykind: MultiMap<Sensor, usize>,
+
+    // list of all devices and a list of their sensors, with an optional sensor
+    // name (if present)
+    device_sensors: Vec<Vec<(Option<String>, Sensor, usize)>>,
```

Comment:
This tuple could also be a struct, since it's not clear what the `usize` does

Reply:
Rolled in with the others in 5bf5ae8b